miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryDilutionService.java

```diff
@@ -0,0 +1,246 @@
+    Collection<LibraryDilution> allDilutions = dilutionDao.listAllLibraryDilutionsByPlatform(platform);
+    return authorizationManager.filterUnreadable(allDilutions);
+  }
+
+  @Override
+  public List<LibraryDilution> getAllByProjectId(Long projectId) throws IOException {
+    Collection<LibraryDilution> allDilutions = dilutionDao.listAllLibraryDilutionsByProjectId(projectId);
+    return authorizationManager.filterUnreadable(allDilutions);
+  }
+
+  @Override
+  public List<LibraryDilution> getAllByProjectIdAndPlatform(Long projectId, PlatformType platform) throws IOException {
+    Collection<LibraryDilution> allDilutions = dilutionDao.listAllLibraryDilutionsByProjectAndPlatform(projectId, platform);
+    return authorizationManager.filterUnreadable(allDilutions);
+  }
+
+  @Override
+  public LibraryDilution getByBarcode(String barcode) throws IOException {
+    LibraryDilution dilution = dilutionDao.getLibraryDilutionByBarcode(barcode);
+    return (authorizationManager.readCheck(dilution) ? dilution : null);
+  }
+
+  @Override
+  public List<LibraryDilution> getAllByPageSizeSearchAndPlatform(int offset, int size, String querystr, PlatformType platform,
+      String sortDir, String sortCol) throws IOException {
+    Collection<LibraryDilution> allDilutions = dilutionDao.listBySearchOffsetAndNumResultsAndPlatform(offset, size, querystr, sortDir,
+        sortCol, platform);
+    return authorizationManager.filterUnreadable(allDilutions);
+  }
+
+  /**
+   * Loads persisted objects into LibraryDilution fields. Should be called before saving LibraryDilutions. Loads all member objects
+   * <b>except</b>
+   * <ul>
+   * <li>creator/lastModifier User objects</li>
+   * </ul>
+   * 
+   * @param libraryDilution the LibraryDilution to load entities into. Must contain at least the IDs of objects to load (e.g. to load the
+   *          persisted Library
+   *          into dilution.library, dilution.library.id must be set)
+   * @throws IOException
+   */
+  private void loadChildEntities(LibraryDilution dilution) throws IOException {
+    if (dilution.getLibrary() != null) {
+      dilution.setLibrary(libraryDao.get(dilution.getLibrary().getId()));
+    }
+    if (dilution.getTargetedSequencing() != null) {
+      dilution.setTargetedSequencing(targetedSequencingDao.get(dilution.getTargetedSequencing().getId()));
+    }
+    dilution.setDilutionCreator(authorizationManager.getCurrentUsername());
+  }
+
+  /**
+   * Copies modifiable fields from the source LibraryDilution into the target LibraryDilution to be persisted
+   * 
+   * @param target the persisted LibraryDilution to modify
+   * @param source the modified LibraryDilution to copy modifiable fields from
+   * @throws IOException
+   */
+  private void applyChanges(LibraryDilution target, LibraryDilution source) {
+    target.setConcentration(source.getConcentration());
+    if (!autoGenerateIdBarcodes) target.setIdentificationBarcode(source.getIdentificationBarcode());
```

Comment:
I had assumed that since there's no UI for the user to enter a barcode when idBarcode generation is enabled, there would be no reason to set it. Is that thinking too narrowly?